src/wallet/test/coinselection_tests.cpp

```diff
@@ -151,6 +151,41 @@ BOOST_AUTO_TEST_CASE(bnb_test)
     AddCoins(clone_pool, {2 * CENT, 7 * CENT, 7 * CENT});
     AddDuplicateCoins(clone_pool, 50'000, 5 * CENT);
     TestBnBSuccess("Skip equivalent input sets", clone_pool, /*selection_target=*/16 * CENT, /*expected_input_amounts=*/{2 * CENT, 7 * CENT, 7 * CENT});
+
+    /* Test BnB attempt limit (`TOTAL_TRIES`)
```

Comment:
It would be nice if BnB returned the `total_tries`, like coin_grinder.  I recently added that to the rust implementation since I think it's useful for regression testing.  If BnB was ever refactored or had changes it would come in handy imo.

Reply:
That seems like a good idea, but that’s out of scope for this PR. I have had a refactor of BnB on my todo list for a long time, and it would be part of that, probably.

Reply:
Done here: https://github.com/bitcoin/bitcoin/pull/32150/commits/e2a4d86131e8f1661eb450c50b8072147965b111